src/components/NewCommentForm.tsx

```
@@ -1,6 +1,107 @@
+import classNames from 'classnames';
+import React, { FC, useState } from 'react';
+import { addPostComment } from './services/comments';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  addComment: (comment: Comment) => void;
+  activePostId: number;
+};
+
+export const NewCommentForm: FC<Props> = ({
+  addComment = () => {},
+  activePostId,
+}) => {
+  const [inputName, setInputName] = useState('');
+  const [inputEmail, setInputEmail] = useState('');
+  const [inputText, setInputText] = useState('');
+
+  const [hasInputNameError, setHasInputNameError] = useState(false);
+  const [hasInputEmailError, setHasInputEmailError] = useState(false);
+  const [hasInputTextError, setHasInputTextError] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function resetForm() {
+    setInputText('');
+  }
+
+  function clearForm() {
+    setInputText('');
+    setInputEmail('');
+    setInputName('');
+
+    setHasInputNameError(false);
+    setHasInputEmailError(false);
+    setHasInputTextError(false);
+  }
+
+  const handleSubmit = async (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+
+    if (inputName.trim().length === 0) {
+      setHasInputNameError(true);
+    }
+
+    if (inputEmail.trim().length === 0) {
+      setHasInputEmailError(true);
+    }
+
+    if (inputText.trim().length === 0) {
+      setHasInputTextError(true);
+    }
+
+    if (
+      inputText.trim().length === 0 ||
+      inputEmail.trim().length === 0 ||
+      inputName.trim().length === 0
+    ) {
+      return;
+    } else {
+      const message = {
+        name: inputName,
+        email: inputEmail,
+        body: inputText,
+      };
+
+      const newComment = { ...message, postId: activePostId };
+
+      setIsLoading(true);
+      addPostComment(newComment)
+        .then(comment => {
+          addComment(comment);
+          resetForm();
+        })
+        .catch(() => {})
```

The promise chain swallows errors: `.catch(() => {})`. This silently ignores failures from addPostComment. The checklist (optional) asks to: "Handle add and delete errors so the user can retry." At minimum, surface the error (set an error state or show a notification) and provide a retry or let the parent know the add failed. Don't leave an empty catch block.